pkg/test/case.go

```diff
@@ -94,6 +94,14 @@ func (t *Case) CreateNamespace(cl client.Client, ns *namespace) error {
 		defer cancel()
 	}
 
+	if !t.SkipDelete {
+		test.Cleanup(func() {
+			if err := t.DeleteNamespace(cl, ns); err != nil {
+				test.Error(err)
+			}
+		})
+	}
+
```

Comment:
This looks good, we now can register ns cleanup at the same place where we create it.